src/model/mod.rs

```diff
@@ -17,7 +17,127 @@ pub mod show;
 pub mod track;
 pub mod user;
 
-use serde::{Deserialize, Serialize};
+use chrono::{DateTime, NaiveDateTime, Utc};
+use serde::{de, Deserialize, Serialize, Serializer};
+use std::{fmt, time::Duration};
+
+/// Vistor to help deserialize duration represented as millisecond to `std::time::Duration`
+struct DurationVisitor;
+impl<'de> de::Visitor<'de> for DurationVisitor {
+    type Value = Duration;
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        write!(formatter, "a milliseconds represents std::time::Duration")
+    }
+    fn visit_u64<E>(self, v: u64) -> Result<Self::Value, E>
+    where
+        E: de::Error,
+    {
+        Ok(Duration::from_millis(v))
+    }
+}
+
+/// Deserialize `std::time::Duration` from millisecond(represented as u64)
```

Comment:
These comments could have an extra space before the parenthesis. `millisecond` could be changed to `milliseconds` as well.